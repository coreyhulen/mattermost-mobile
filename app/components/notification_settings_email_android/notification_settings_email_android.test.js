// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import NotificationSettingsEmailAndroid from './notification_settings_email_android.js';

describe('NotificationSettingsEmailAndroid', () => {
    const baseProps = {
        actions: {
            savePreferences: () => {}, // eslint-disable-line no-empty-function
        },
        currentUserId: 'current_user_id',
        emailInterval: '30',
        enableEmailBatching: true,
        onClose: () => {}, // eslint-disable-line no-empty-function
        sendEmailNotifications: true,
        siteName: 'Mattermost',
        theme: {
            centerChannelBg: '#aaa',
            centerChannelColor: '#aaa',
        },
        visible: true,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <NotificationSettingsEmailAndroid {...baseProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should macth state on setEmailNotifications', () => {
        const wrapper = shallow(
            <NotificationSettingsEmailAndroid {...baseProps}/>
        );

        wrapper.setState({email: 'false', interval: '0'});
        wrapper.instance().setEmailNotifications('30');
        expect(wrapper.state({email: 'true', interval: '30'}));

        wrapper.instance().setEmailNotifications('0');
        expect(wrapper.state({email: 'false', interval: '0'}));

        wrapper.instance().setEmailNotifications('3600');
        expect(wrapper.state({email: 'true', interval: '3600'}));
    });

    test('should call props.onClose on handleClose', () => {
        const props = {...baseProps, onClose: jest.fn()};
        const wrapper = shallow(
            <NotificationSettingsEmailAndroid {...props}/>
        );

        wrapper.instance().handleClose();
        expect(props.onClose).toHaveBeenCalledTimes(1);
        expect(props.onClose).toBeCalledWith();
    });

    test('should call props.actions.savePreferences and props.onClose on handleSaveEmailNotification', () => {
        const props = {...baseProps, onClose: jest.fn(), actions: {savePreferences: jest.fn()}};
        const wrapper = shallow(
            <NotificationSettingsEmailAndroid {...props}/>
        );

        wrapper.setState({email: 'true', interval: '3600'});
        wrapper.instance().handleSaveEmailNotification();
        expect(props.actions.savePreferences).toHaveBeenCalledTimes(1);
        expect(props.actions.savePreferences).toBeCalledWith(
            'current_user_id',
            [
                {category: 'notifications', name: 'email', user_id: 'current_user_id', value: 'true'},
                {category: 'notifications', name: 'email_interval', user_id: 'current_user_id', value: '3600'},
            ]
        );
        expect(props.onClose).toHaveBeenCalledTimes(1);
        expect(props.onClose).toBeCalledWith();
    });
});
