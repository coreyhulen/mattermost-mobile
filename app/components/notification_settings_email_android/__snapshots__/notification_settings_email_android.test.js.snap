// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettingsEmailAndroid should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationSettingsEmailAndroid
    actions={
      Object {
        "savePreferences": [Function],
      }
    }
    currentUserId="current_user_id"
    emailInterval="30"
    enableEmailBatching={true}
    onClose={[Function]}
    sendEmailNotifications={true}
    siteName="Mattermost"
    theme={
      Object {
        "centerChannelBg": "#aaa",
        "centerChannelColor": "#aaa",
      }
    }
    visible={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
